gui/src/graphics/mod.rs

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+#[cfg_attr(target_os = "macos", path = "metal.rs")]
+#[cfg_attr(not(target_os = "macos"), path = "vulkan.rs")]
+mod api;
+
+#[cfg(not(target_os = "macos"))]
+pub type DefaultApi = self::api::Vulkan;
+
+#[cfg(target_os = "macos")]
+pub type DefaultApi = self::api::Metal;
+
+pub trait GraphicsApi: Sized + 'static {
+    type PhysicalDevice: PhysicalDevice;
+
+    type CreateError: core::error::Error;
+
+    fn new() -> Result<Self, Self::CreateError>;
+
+    fn enumerate_physical_devices(&self) -> &[Self::PhysicalDevice];
```

Comment:
The method name tell it is going to enumerate the devices while the actual behavior is returned a list of devices that already enumerated so better to rename to just `physical_devices`.

Reply:
good point